File: Exercism/typescript/atbash-cipher/atbash-cipher.ts
const ALPHABET: string = 'abcdefghijklmnopqrstuvwxyz0123456789'
const CIPHER: string = 'zyxwvutsrqponmlkjihgfedcba0123456789'
const PUNCT_AND_NUMS_REGEX = /[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-./:;<=>?@[\]^_`{|}~]/g;

/**
 * Atbash ciphers a string. Adds space character every 5 characters for extra security.
 * @param s String to be encoded.
 * @returns Encoded string.
 */
export function encode(s: string): string {
  return atbash(s, ALPHABET, CIPHER).replace(/(.{5})/g, '$1 ').trim()
}

/**
 * Decodes Atbash encoded string.
 * @param s String to be decoded.
 * @returns Decoded string.
 */
export function decode(s: string): string {
  return atbash(s, CIPHER, ALPHABET)
}
/**
 * Map string characters from one alphabet to another.
 * @param s String to be mapped.
 * @param from Alphabet that string s is written in.
 * @param to Alphabet to map to.
 * @returns Mapped string.
 */
function atbash(s: string, from: string, to: string): string {
  return s
    .replace(PUNCT_AND_NUMS_REGEX, '')
    .toLowerCase()
    .split('')
    .map((el) => {
      return to[from.indexOf(el)]
    })
    .join('')
}

File: Exercism/cpp/atbash-cipher/atbash_cipher.cpp
#include "atbash_cipher.h"

namespace atbash_cipher {
  const string ALPHABET = "abcdefghijklmnopqrstuvwxyz0123456789";
  const string CIPHER = "zyxwvutsrqponmlkjihgfedcba0123456789";

  string encode(string _s) {
    return replace(_s, CIPHER, ALPHABET, true);
  }

  string decode(string _s) {
    return replace(_s, ALPHABET, CIPHER, false);
  }

  string replace(string _what, string _from, string _to, bool _seperate) {
    transform(_what.begin(), _what.end(), _what.begin(), ::tolower);
    _what = regex_replace(_what, regex(R"([,|.| ])"), "");

    string ret;

    for(unsigned int i = 0; i < _what.length(); i++) {
      if(_seperate && i != 0 && i % 5 == 0) {
        ret += " ";
      }

      ret += _to[_from.find(_what[i])];
    }

    return ret;
  }
}  // namespace atbash_cipher
